Feature/cronjob

jbarbaso opened the pull request:
Added status field, setters and getters to Car entity class.
Added named query to get all the inactive cars in database.
Created ActivateCarTimer to the project.

jbarbaso committed:
Added status field, setters and getters to Car entity class.
Added named query to get all the inactive cars in database.

jbarbaso committed:
Created ActivateCarTimer to the project.

javieraviles commented on src/main/java/com/everis/cars/timer/ActivateCarTimer.java:
```diff
@@ -0,0 +1,76 @@
+	 * 
+	 * @see javax.persistence.EntityManager
+	 */
+	@PersistenceContext(unitName = "em_postgres")
```
this should not be here, only PersistenceService should inject persistenceContext

jbarbaso replied:
Removed PersistenceService from ActivateCarTimer.

javieraviles commented on src/main/java/com/everis/cars/timer/ActivateCarTimer.java:
```diff
@@ -0,0 +1,76 @@
+	/**
+	 * @return a {@link Car} {@link List} with false status value
+	 */
+	private List<Car> getInactiveCars() {
```
this method should come from CarService. On top of that, carService should delegate the execution of the namedquery to PersistenceService.

javieraviles replied:
something like this inside PersistenceService?

public <T> List<T> findWithNamedQuery(final String namedQueryName) {
	return entityManager.createNamedQuery(namedQueryName).getResultList();
}

jbarbaso replied:
Created a method to execute named queries from PersistenceService.
Moved getInactiveCars method from ActivateCarTimer to CarService.

jbarbaso committed:
Created a method to execute named queries from PersistenceService.
Moved getInactiveCars method from ActivateCarTimer to CarService.
Removed PersistenceService from ActivateCarTimer.
Cleaned useless dependencies and formated code.

jbarbaso commented:
Comments solved.
